icu4c/source/tools/toolutil/toolutil.cpp

```diff
@@ -218,6 +219,37 @@ uprv_fileExists(const char *file) {
 }
 #endif
 
+U_CAPI int32_t U_EXPORT2
+uprv_compareGoldenFiles(
+        const char* buffer, int32_t bufferLen,
+        const char* goldenFilePath,
+        bool overwrite) {
+
+    if (overwrite) {
+        std::ofstream ofs;
+        ofs.open(goldenFilePath);
+        ofs.write(buffer, bufferLen);
+        ofs.close();
+    }
+
+    std::ifstream ifs(goldenFilePath, std::ifstream::in);
+    char c = ifs.get();
```

Comment:
Why do you "get" the first byte before the loop?
It would be nice to only have one "get", and that would make it work for empty files, too.

(Together with the next comment -- loop while not EOF, and get inside the loop.)

Reply:
Done